src/pages/workspaces/workspace/workflows/WorkflowView.js

```diff
@@ -951,6 +951,27 @@ const WorkflowView = _.flow(
         _.update('outputs', this.isSingle() ? () => ({}) : _.mapValues(_.trim))
       )
 
+      // Dockstore users who target floating tags can change their WDL via Github without explicitly selecting a new version in Terra.
+      // If the change removes or renames inputs or outputs, the `modifiedConfig` retrieved from the DB will include stale keys that
+      // are hidden from the UI and cause the workflow to fail at launch with "Validation errors: Extra inputs".
+      //
+      // The IO set `modifiedInputsOutputs` is always current because we regenerate it from the WDL each time; use it to filter out
+      // any extraneous IO keys before saving.
+      //
+      // https://broadworkbench.atlassian.net/browse/WA-291
+      const validInputs = _.map('name', modifiedInputsOutputs['inputs'])
+      const validOutputs = _.map('name', modifiedInputsOutputs['outputs'])
+
+      const filterIO = (ioMap, allowedKeys) => {
+        return _.fromPairs(_.filter(pair => {
+          return _.includes(pair[0], allowedKeys)
+        }, _.toPairs(ioMap)))
+      }
+
+      modifiedConfig['inputs'] = filterIO(modifiedConfig['inputs'], validInputs)
+      modifiedConfig['outputs'] = filterIO(modifiedConfig['outputs'], validOutputs)
+
+      // POST to create new MC snapshot
```

Comment:
TOL: Instead of doing this on save, we could do it on initial load, e.g. in `componentDidMount`. The difference would be that the Save button would be enabled from the start, giving a soft indication that something has changed (and also giving us the opportunity to explicitly call this out with a warning in the future, if we want)

Reply:
I just thought of a much stronger argument for this suggestion: Due to existing UI logic, filtering on load would force the user to save before being allowed to launch analysis, which should prevent the invalid fields from being sent to cromwell.